equinox/nn/_stateful.py

```diff
@@ -145,3 +146,24 @@ def tree_unflatten(cls, keys, values):
             state[key] = value
         self._state = state
         return self
+
+
+class StatefulLayer(Module):
+    """An abstract base class representing a stateful layer within a neural network.
+
+    Stateful layers maintain internal state across calls, which can be useful for tasks
+    such as BatchNormalisation.
+
+    Subclasses must implement the `__call__` method that takes input data and the
+    current state as arguments and returns the output data and updated state.
+    """
```

Comment:
Can you (a) mention that right now this is used solely to declare the API for use with Sequential, and (b) add this base class to the documentation?